controllers/baremetalset_controller.go

```diff
@@ -274,32 +274,67 @@ func (r *BaremetalSetReconciler) ensureBaremetalHosts(instance *ospdirectorv1bet
 		return err
 	}
 
-	existingBaremetalHosts := []string{}
+	existingBaremetalHosts := map[string]string{}
+	removalAnnotatedBaremetalHosts := []string{}
 
 	// Find the current BaremetalHosts belonging to this BaremetalSet
 	for _, baremetalHost := range baremetalHostsList.Items {
 		if baremetalHost.Spec.ConsumerRef != nil && (baremetalHost.Spec.ConsumerRef.Kind == instance.Kind && baremetalHost.Spec.ConsumerRef.Name == instance.Name && baremetalHost.Spec.ConsumerRef.Namespace == instance.Namespace) {
 			r.Log.Info(fmt.Sprintf("Existing BaremetalHost: %s", baremetalHost.ObjectMeta.Name))
-			existingBaremetalHosts = append(existingBaremetalHosts, baremetalHost.ObjectMeta.Name)
+			// Storing the existing BaremetalHosts in a map like this just makes selected replica scale-down easier below
+			existingBaremetalHosts[baremetalHost.ObjectMeta.Name] = baremetalHost.ObjectMeta.Name
+
+			// Prep for possible replica scale-down below
+			if val, ok := baremetalHost.Annotations[baremetalset.BaremetalHostRemovalAnnotation]; ok && (val == "yes" || val == "true") {
```

Comment:
we probably should use https://golang.org/pkg/strings/#EqualFold or strings.ToLower(val) to do case-insensitivity  compare, just to make sure someone did not set BaremetalHostRemovalAnnotation to Yes/True?

Reply:
Sure, that's a good point